File: js/algorithm/bst/avl.js
let BST = require("./bst")
let AVLNode = require("./avl_node")

class AVL extends BST {
  constructor(data) {
    super()
    this.root = null
    if (typeof(data) === 'number') {
      this.root = new AVLNode(data)
    } else if (Array.isArray(data)) {
      this.root = new AVLNode(data[0])
      for (let i = 1; i < data.length; i++)
        this.insert(data[i])
    }
  }
}

module.exports = AVL

File: python/algorithm/bst/avl.py
from bst import *
from avlnode import *

class AVL(BST):
    def __init__(self, data):
        self.root = None
        if isinstance(data, int):
            self._root = AVLNode(data)
        elif isinstance(data, list):
            self._root = AVLNode(data[0])
            for i in range(1, len(data), 1):
                self.insert(data[i])

    def insert(self, val):
        print(">> insert", val)
        if self._root is None:
            self._root = AVLNode(val)

        self._root = self._root.insert(val)
